File: web-admin/src/types/contractTemplate.ts
export interface ContractTemplate {
    id: string;
    name: string;
    description?: string;
    header_content?: string;
    body_content: string;
    footer_content?: string;
    is_active: boolean;
    created_at?: string;
    updated_at?: string;
}

export interface CreateContractTemplateRequest {
    name: string;
    description?: string;
    header_content?: string;
    body_content: string;
    footer_content?: string;
}

export interface UpdateContractTemplateRequest {
    name?: string;
    description?: string;
    header_content?: string;
    body_content?: string;
    footer_content?: string;
    is_active?: boolean;
}

File: crates/core/src/domain/entities/contract_template.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct ContractTemplate {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub header_content: Option<String>,
    pub body_content: String,
    pub footer_content: Option<String>,
    pub is_active: bool,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
}
